preptime/routes/users.js

```diff
@@ -0,0 +1,220 @@
+const express = require("express");
+const router = express.Router();
+
+router.get("/", (req, res) => {
+  res.send("USERS API IS RUNNING");
+});
+
+router.get("/userInfo/:userId", async (req, res) => {
+  const admin = req.app.locals.admin;
+  const db = admin.firestore();
+
+  const userId = req.params.userId;
+
+  try {
+    // Get user information from Firestore based on userId
+    const userDoc = await db.collection("users").doc(userId).get();
+
+    if (!userDoc.exists) {
+      return res.status(404).send({ error: "User not found" });
+    }
+
+    const userData = userDoc.data();
+
+    res.send({
+      fullName: userData.fullName || "FULL NAME",
+      profileImg: userData.profileImg || "IMAGE",
+      followers: userData.followers || [],
+      following: userData.followingUsers || [],
+      favoriteTeams: userData.favoriteTeams || [],
+    });
+  } catch (error) {
+    console.error("Error fetching user data:", error);
+    res.status(500).send({ error: "Internal server error" });
+  }
+});
+
+router.post("/followUser", async (req, res) => {
+  const admin = req.app.locals.admin;
+  const db = admin.firestore();
+
+  // Check if req.body is defined
+  if (!req.body) {
+    return res.status(400).send({ error: "Request body is missing" });
+  }
+
+  const { userToFollowId, currentUserId } = req.body;
```

Comment:
Here and in a few other routes the currentUserId is passed in the body. Should it be retrieved a different way when login works? You can solve this in a later commit